gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -89,6 +89,130 @@ def _load_cadd_raw(cadd_tsv) -> hl.Table:
     return ht
 
 
+def create_spliceai_grch38_ht(
+    snvs_path: str, indels_path: str, new_indels_path: str, header_file_path: str
+) -> hl.Table:
+    """
+    Create a Hail Table with SpliceAI scores for GRCh38.
+
+    ..Note::
+    The SpliceAI scores are from the following resources:
+    all SNVs precomputed: spliceai_scores.masked.snv.hg38.vcf.bgz, downloaded from https://basespace.illumina.com/s/5u6ThOblecrh
+    all indels precomputed: spliceai_scores.masked.indel.hg38.vcf.bgz, downloaded from https://basespace.illumina.com/s/5u6ThOblecrh
+    new indels from gnomAD v4: gnomad_v4_new_indels.spliceai_masked.vcf.bgz, requested Illumina to compute for us.
+    We may have lost the SpliceAI score for gnomAD v3 indels, which makes it hard to process the raw scores to get a maximum and retain the position information.
+    Before we didn't `explode` the SpliceAI scores, so we only had scores for the first gene for each variant.
+
+    :return: Hail Table with SpliceAI scores for GRCh38.
+    """
+    logger.info("Importing vcf of SpliceAI scores into MT...")
+    recode = {f"{i}": f"chr{i}" for i in (list(range(1, 23)) + ["X", "Y"])}
+    spliceai_snvs = hl.import_vcf(
+        snvs_path,
+        force_bgz=True,
+        min_partitions=1000,
+        reference_genome="GRCh38",
+        contig_recoding=recode,
+        skip_invalid_loci=True,
+    )
+
+    spliceai_indels = hl.import_vcf(
+        indels_path,
+        force_bgz=True,
+        reference_genome="GRCh38",
+        contig_recoding=recode,
+        skip_invalid_loci=True,
+        min_partitions=1000,
+    )
+
+    spliceai_new_indels = hl.import_vcf(
+        new_indels_path,
+        header_file=header_file_path,
+        force_bgz=True,
+        reference_genome="GRCh38",
+        contig_recoding=recode,
+        skip_invalid_loci=True,
+        min_partitions=1000,
+    )
+
+    mt = spliceai_snvs.union_rows(spliceai_indels).union_rows(spliceai_new_indels)
+    ht = mt.rows()
+
+    logger.info("Exploding SpliceAI scores...")
+    # `explode` will eliminate rows with empty array, but the varaints with multiple genes will extend the number of rows.
```

Comment:
this is because SpliceAI only has predictions for variants inside the gene body, so number of lines != number of variants with score(s), I will put a note here.